src/main/java/seedu/address/logic/parser/SortCommandParser.java

```diff
@@ -13,18 +17,35 @@
 public class SortCommandParser implements Parser<SortCommand> {
 
     /**
-     * Parses the given {@code String} of arguments in the context of the FindCommand
-     * and returns a FindCommand object for execution.
+     * Parses the given {@code String} of arguments in the context of the SortCommand
+     * and returns a SortCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public SortCommand parse(String args) throws ParseException {
+
         String trimmedArgs = args.toLowerCase().trim();
-        switch (trimmedArgs) {
-        case "name":
-            return new SortCommand(Applicant.sortByName());
-        case "scholarship":
-            return new SortCommand(Applicant.sortByScholarship());
-        default:
+        String[] keywords = trimmedArgs.split("\\s+");
+        List<String> keywordsList = Arrays.asList(keywords);
+
+        if (keywordsList.contains("name")) {
+            return new SortCommand(checkInputSizeAndReverseFlag(Applicant.sortByName(), keywordsList));
+        } else if (keywordsList.contains("scholarship")) {
+            return new SortCommand(checkInputSizeAndReverseFlag(Applicant.sortByScholarship(), keywordsList));
+        } else if (keywordsList.contains("status")) {
+            return new SortCommand(checkInputSizeAndReverseFlag(Applicant.sortByStatus(), keywordsList));
+        } else {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        }
+    }
+
+    private static Comparator<Applicant> checkInputSizeAndReverseFlag(
+            Comparator<Applicant> comparator, List<String> list) throws ParseException {
+        if (list.size() == 2 && list.contains("-r")) {
+            return comparator.reversed();
+        } else if (list.size() == 1) {
```

Comment:
Problem of magic strings "name", "scholarship", ... and "-r". Maybe can specify them somewhere?

Reply:
Noted, will make amendments.